python/taichi/lang/matrix.py

```diff
@@ -497,11 +497,17 @@ def assign_renamed(x, y):
         from .meta import fill_matrix
         fill_matrix(self, val)
 
+    def shape_ext(self, as_vector=None):
```

Comment:
It's an off-topic change: some GAMES 201 students complains that old codes with `as_vector=True` cause an error. So let's make it a deprecation warning instead of throwing an error.